jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunMojo.java

```diff
@@ -308,15 +315,86 @@ public void configureWebApplication() throws Exception
                 }
             }
         }
+        List<Overlay> warProjectConfiguration = getWarProjectConfiguration();
+        configureWebInf(warProjectConfiguration);
+        configureBaseResource(warProjectConfiguration);
 
-        //process any overlays and the war type artifacts
-        List<Overlay> overlays = getOverlays();
-        unpackOverlays(overlays); //this sets up the base resource collection
+        // this is to handle WARs that are not referenced in maven reactor
+        List<Overlay> overlaysToUnpack = warProjectConfiguration.stream()
+            .filter(e -> !e.isMavenProject())
+            .collect(Collectors.toList());
+        unpackOverlays(overlaysToUnpack);
 
         getLog().info("web.xml file = " + webApp.getDescriptor());
         getLog().info("Webapp directory = " + webAppSourceDirectory.getCanonicalPath());
     }
 
+    private void configureWebInf(List<Overlay> warProjectConfiguration)
+        throws IOException
+    {
+        Set<File> webInfLib = new LinkedHashSet<>();
+        Set<File> webInfClasses = new LinkedHashSet<>();
+        for (Overlay overlay : warProjectConfiguration)
+        {
+            Artifact artifact = overlay.getArtifact();
+            MavenProject project = mavenProjectHelper.getMavenProject(artifact);
+            if (project != null)
+            {
+                getLog().debug("WEB-INF/lib of " + artifact.getArtifactId() + " resolved from maven reactor");
+                getWebInfLibArtifacts(project).stream()
+                    .map(mavenProjectHelper::getArtifactPath)
+                    .map(Path::toFile)
+                    .forEach(webInfLib::add);
+                webInfClasses.add(mavenProjectHelper.getArtifactPath(artifact).toFile());
+            }
+            else
+            {
+                getLog().debug("WEB-INF/lib of " + artifact.getArtifactId() + " will be unpacked to target jetty_overlays");
+            }
+        }
+        webInfClasses.remove(webApp.getClasses());
+        webInfClasses.remove(webApp.getTestClasses());
```

Comment:
Why are the current classes and test classes removed?

Reply:
they are already add in JettyRunMojo:277 and JettyRunMojo:281, i find in cleaner to just remove them once all is add, instead of doing a check each time i am trying to add something

Reply:
Better not to add them in the first place if they're not needed?

Reply:
I will take a look at the implementation and see what actually happens there again - I don't remember from top of my head